### PR TITLE
[DNM] Synch 29 07 2021

### PR DESCRIPTION
**Pull request by nvlsianpu**
Synchronize up to https://github.com/mcu-tools/mcuboot/commit/994816d04a055b39de2d174930271023965e8138

Merge using **Merge by Create a Merge commit!**

### CODE REVIEW

**Review by mbolivar-nordic — APPROVED** [review]

**Comment by de-nordic** [comment]
I have tested the sync on nrf52840dk_nrf52840. The serial recovery seems to work correctly. I have found two issues, that seem to be present on current main also:
 - https://github.com/mcu-tools/mcuboot/issues/1092
 - https://github.com/mcu-tools/mcuboot/issues/1091

As above issues already exist on main, they should not affect the sync.

**Comment by de-nordic** [comment]
It is worth to also take bd69c3dd4b4b7cd0c8063c2d3a885c0664962bc7 with the sync, this is commit directly above sync ending commit but it enables non-peruser mcumgr commands to be handled by platform specific handler.

File: boot/boot_serial/src/boot_serial.c [hunk 1]
```diff
@@ -483,6 +504,11 @@ boot_serial_input(char *buf, int len)
         default:
             break;
         }
+    } else if (MCUBOOT_PERUSER_MGMT_GROUP_ENABLED == 1 &&
+               hdr->nh_group >= MGMT_GROUP_ID_PERUSER) {
```
Comment (de-nordic) [inline boot/boot_serial/src/boot_serial.c:508]: Worth taking https://github.com/zephyrproject-rtos/mcuboot/commit/bd69c3dd4b4b7cd0c8063c2d3a885c0664962bc7 in, that removes this line.

**Review by de-nordic — APPROVED** [review]
Looks OK.

I have tested most common functionalities on nrf52840dk, and everything seems to work OK.
Allowed myself to pull in one more commit: bd69c3dd4b4b7cd0c8063c2d3a885c0664962bc7, moving the Sync head up by one notch.

**Review by mnkp — CHANGES_REQUESTED** [review]
#mcu-tools/mcuboot/pull/1040 introduced a regression, as a result MCUboot does not build on the following boards: 96b_carbon, hexiwear_k64, nucleo_f401re, stm32f0_disco.

**Comment by nvlsianpu** [comment]
^^ pending fix https://github.com/mcu-tools/mcuboot/issues/1101

**Comment by nvlsianpu** [comment]
Awaiting for https://github.com/mcu-tools/mcuboot/pull/1101
Synchronization will be reapplied again to the upstream SHA with that patch.

**Comment by nvlsianpu** [comment]
@mnkp I moved synchronization point ahead so this contains fix for the issue you had mentioned now.

**Review by mnkp — APPROVED** [review]

**Comment by utzig** [comment]
@nvlsianpu You should just disable the CI for this repo, there's no point in running the same upstream tests again...

**Review by utzig — APPROVED** [review]

**Comment by carlescufi** [comment]
@d3zd3z there was this upmerge before you posted yours, so we probably need to close this one now.

**Comment by d3zd3z** [comment]
I'm fine with closing this one. Mine is one commit behind, which includes the release notes for 1.8.0-rc1, which we can include if we want.

**Pull request closed without merging**